swift/Arrow/Sources/Arrow/ArrowCExporter.swift

```diff
@@ -0,0 +1,110 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import Foundation
+import ArrowC
+import Atomics
+
+extension String {
+    var cstring: UnsafePointer<CChar> {
+        (self as NSString).cString(using: String.Encoding.utf8.rawValue)!
```

Comment:
Why not https://developer.apple.com/documentation/swift/string/utf8cstring or https://developer.apple.com/documentation/swift/string/withcstring(_:) ?

Reply:
withcstring and withUnsafeBufferPointer: The pointer passed as an argument to body is valid only during the execution of withCString(_:). Do not store or return the pointer for later use. 
 
 We cannot use those for this reason.

Reply:
But you have the same kind of issue here:
https://developer.apple.com/documentation/foundation/nsstring/1408489-cstring#discussion

Specifically:
> The returned C string is guaranteed to be valid only until either the receiver is freed, or until the current memory is emptied, whichever occurs first. You should copy the C string or use [getCString(_:maxLength:encoding:)](https://developer.apple.com/documentation/foundation/nsstring/1415702-getcstring) if it needs to store the C string beyond this time.

Reply:
I have changed the code to store the schema exported data that is similar to the export array so this cstring will remain in memory until released.